src/model.py

```diff
@@ -50,6 +57,64 @@ def forward(self, inputs: torch.Tensor) -> torch.Tensor:
         return self.out(x)
 
 
+class ResidualBottleNeck(nn.Module):
```

Comment:
Could we rename this class as `BottleneckResidualBlock`? I know the textbook uses this name, but it just sounds like it's referring to the bottleneck layers *within* a block.